pkg/build/api/v1/types.go

```diff
@@ -163,6 +163,9 @@ type BuildSource struct {
 	// repository.
 	ContextDir string `json:"contextDir,omitempty" description:"specifies sub-directory where the source code for the application exists, allows for sources to be built from a directory other than the root of a repository"`
 
+	// DockerfileName is the name of the Dockerfile that will be used to build the Docker image.
+	DockerfileName string `json:"dockerfileName,omitempty" description:"specifies the name of the Dockerfile to use for building the Docker image"`
```

Comment:
if we're going to go down the path you suggest, then it seems like it would make more sense to get rid of the generic BuildSource entirely, and instead define a XXXBuildSource type for each type of build.

There would be overlap between them of course (eg all build sources would support a git location field).

if we're not going to do that, i think we should continue down the path we've established of having a single generic buildsource object which contains the superset of inputs that our strategies can understand, and handle invalid values via validation.

but saying that "Dockerfile name" is not an input because it's not supported by all build strategies is ignoring the other fields that already exist in the object.